server/server.go

```diff
@@ -219,9 +214,11 @@ func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
 		go func() {
 			for range time.Tick(time.Hour * 24 * 30) { // 30 days
 				logutil.BgLogger().Info("Rotating automatically created TLS Certificates")
-				tlsConfig, _, err = util.LoadTLSCertificates(
-					s.cfg.Security.SSLCA, s.cfg.Security.SSLKey, s.cfg.Security.SSLCert,
-					s.cfg.Security.AutoTLS, s.cfg.Security.RSAKeySize)
+				tlsConfig, err = util.LoadTLSCertificates(
+					variable.GetSysVar("ssl_ca").Value,
+					variable.GetSysVar("ssl_key").Value,
+					variable.GetSysVar("ssl_cert").Value,
+				)
```

Comment:
These TLS changes seem unrelated. I don't think there's any issue, but would you mind moving them to a different PR?

Reply:
Before, the config items and sysvars related to ssl_key and ssl_cert is unchangeable. If `auto-tls` is enable but not specifying `ssl-key` or `ssl-cert` in the configuration file, tidb-server would auto generate them in the temp directory, and 'autoReload' them using the same paths every 3 months.

But if the path of temp directory is changeable now, we have to avoid reload different location of ssl_key or ssl_cert after tidb-server starting. So changes are make - generated ssl_key and ssl_cert are recorded in sysvars `ssl_key` and `ssl_cert`, and tidb-server always load them using the same path.

I think it is better to move these snippets to another PR :-)